controller/comments.go

```diff
@@ -112,10 +113,16 @@ func (c *CommentsController) Delete(ctx *app.DeleteCommentsContext) error {
 		if err != nil {
 			return jsonapi.JSONErrorResponse(ctx, err)
 		}
-		if *identityID != cm.CreatedBy {
-			// need to use the goa.NewErrorClass() func as there is no native support for 403 in goa
-			// and it is not planned to be supported yet: https://github.com/goadesign/goa/pull/1030
-			return jsonapi.JSONErrorResponse(ctx, goa.NewErrorClass("forbidden", 403)("User is not the comment author"))
+		wi, err := appl.WorkItems().LoadByID(ctx.Context, cm.ParentID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		authorized, err := authz.Authorize(ctx, wi.SpaceID.String())
```

Comment:
Yes, there are two transactions now. But you still call authz.Authorize(ctx, wi.SpaceID.String()) inside a transaction.
It should be:
```
var cm *comment.Comment
err := application.Transactional(c.db, func(appl application.Application) error {
     // The first transaction to obtain the comment from the DB
}
if *identityID != cm.CreatedBy {
     // Forbidden
}
authorized, err := authz.Authorize(ctx, wi.SpaceID.String())
if !authorized {
    // Forbidden
}
err := application.Transactional(c.db, func(appl application.Application) error {
     // The second transaction to delete the comment
}
```